### PR TITLE
JBEHAVE-1511 Add reporting of hooks

### PR DESCRIPTION
**Pull request by uarlouski**
debug

### CODE REVIEW

File: distribution/src/site/content/migration-paths.html [hunk 1]
```diff
@@ -39,7 +39,8 @@ <h2>From 4.x to 5.0</h2>
     <li>Fix typo in StoryReporter method: lifecyle -> lifecycle</li>
     <li>Replace StoryReporter#beforeStep(String) hook with StoryReporter#beforeStep(Step)</li>
     <li>Invoke StoryReporter#beforeStep(Step) hook before all steps (including comment, pending, ignorable), previously the hook was invoked only for parameterized steps</li>
-    <li> CurrencyConverter, PatternConverter, FileConverter and VerbatimConverter classes were removed, their functionality is covered by new instances of FunctionalParameterConverter class</li>
+    <li>CurrencyConverter, PatternConverter, FileConverter and VerbatimConverter classes were removed, their functionality is covered by new instances of FunctionalParameterConverter class</li>
+    <li>Report StoryReporter.beforeStep(Step) hook and non-silent oucome for methods annotated with @BeforeStories, @AfterStories, @BeforeStory, @AfterStory, @BeforeScenario, @AfterScenario</li>
```
Comment (valfirst) [inline distribution/src/site/content/migration-paths.html]:
```suggestion
    <li>Always trigger StoryReporter.beforeStep(Step) hook and report all outcomes (previously only failures were reported, successful outcome was silent) for methods annotated with @BeforeStories, @AfterStories, @BeforeStory, @AfterStory, @BeforeScenario, @AfterScenario</li>
```

File: jbehave-core/src/main/java/org/jbehave/core/steps/StepCreator.java [hunk 2]
```diff
@@ -731,17 +730,18 @@ public String asString(Keywords keywords) {
         }
     }
 
-    private class BeforeOrAfterStep extends AbstractStep {
+    private class BeforeOrAfterStep extends ReportingAbstractStep {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/steps/StepCreator.java:733]: just to confirm: do we still need both `AbstractStep` and `ReportingAbstractStep`?

File: jbehave-core/src/test/java/org/jbehave/core/JBehaveMatchers.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package org.jbehave.core;
```
Comment (valfirst) [inline jbehave-core/src/test/java/org/jbehave/core/JBehaveMatchers.java]:
```suggestion
package org.jbehave.core.steps;
```
?

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

File: jbehave-core/src/main/java/org/jbehave/core/steps/StepCreator.java (hunk 2 above)
Comment (uarlouski) [reply jbehave-core/src/main/java/org/jbehave/core/steps/StepCreator.java:733]: decided to left it as it is to keep unchanged the string values returned for reporting steps and Upon(Any|Failed|Success)Step that use default implementation

**Review by ikalinin1 — APPROVED** [review]

**Comment by uarlouski** [comment]
In master https://github.com/jbehave/jbehave-core/commit/a3e8f7c3677faeaaa10d1cb5430126334a1faf10

**Pull request closed without merging**